packages/chevrotain/src/parse/parser/parser.ts

```diff
@@ -269,8 +269,9 @@ export class Parser {
       )
     }
 
-    this.skipValidations =
-      config.skipValidations ?? DEFAULT_PARSER_CONFIG.skipValidations
+    this.skipValidations = has(config, "skipValidations")
+      ? (config.skipValidations as boolean) // casting assumes the end user passing the correct type
```

Comment:
The `!` in other places is fine because we know that is actually defined.

I would strongly suggest against incorrect type-assertions. If you really do not want to use the variant I did, then the type of config.skipValidations should be changed to `boolean | undefined`. Then we will ensure that other usages of config.skipValidations take into account the fact that it can be undefined.

Reply:
The a ?? b operator means the same thing as a !== undefined && a !== null ? a : b.

For cases `{}` and `{skipValidations: myBoolean}`, this has no semantic differences. For `{skipValidations: undefined}`, this now also uses DEFAULT_PARSER_CONFIG.skipValidations. Before it would have assigned undefined. If DEFAULT_PARSER_CONFIG.skipValidations is true then there are probably no semantic differences in general.

Reply:
Hi @NaridaL 

It is not that I don't want to use the variant you did.
It is much more modern and concise style of ECMAScript 👍 

It is that:
1. I don't want to change the behavior of the public APIs (even in a minor way) as part of a refactoring PR.
2. I want to remain consistent with the rest of the code base in how it handles config properties which are provided by end users.

I am not opposed to changing it everywhere in the code base, (once we consider pros/cons).
But if we choose to stick with the existing `has` check then I am also fine with the `as boolean` cast because
it describes the author's intent:
- "Responsibility of end-user to provide correct type"